### PR TITLE
SMF failure handling

### PR DESCRIPTION
**Pull request by vthiruveedula**
Pronto-133: SMF is not forwarding/sending PDU Session Establishment Reject to AMF

### CODE REVIEW

**Review by thakurajayL — CHANGES_REQUESTED** [review]

File: context/db.go [hunk 1]
```diff
@@ -305,24 +306,29 @@ func GetSeidByRefInDB(ref string) (seid uint64) {
 }
 
 // GetSMContext By Ref from DB
-func GetSMContextByRefInDB(ref string) (smContext *SMContext) {
+func GetSMContextByRefInDB(ref string) (smContext *SMContext, err error) {
 	fmt.Println("db - in GetSMContextByRefInDB")
```
Comment (UpadhyayaAnkur) [inline context/db.go:310]: Please replace fmt.Println with smf log and ref printed in it.

File: context/db.go [hunk 2]
```diff
@@ -305,24 +306,29 @@ func GetSeidByRefInDB(ref string) (seid uint64) {
...
 	smContext = &SMContext{}
 	filter := bson.M{}
 	filter["ref"] = ref
 
 	result := MongoDBLibrary.RestfulAPIGetOne(SmContextDataColl, filter)
 
-	err := json.Unmarshal(mapToByte(result), smContext)
 	smContext.SMLock.Lock()
```
Comment (UpadhyayaAnkur) [inline context/db.go:317]: Probably, taking this lock is not required. Lets discuss this

File: context/sm_context.go [hunk 3]
```diff
@@ -293,8 +293,10 @@ func GetSMContext(ref string) (smContext *SMContext) {
 		smContext = value.(*SMContext)
 	} else {
 		if factory.SmfConfig.Configuration.EnableDbStore {
-			smContext = GetSMContextByRefInDB(ref)
-			smContextPool.Store(ref, smContext)
+			smContext, _ := GetSMContextByRefInDB(ref)
```
Comment (UpadhyayaAnkur) [inline context/sm_context.go:296]: Error added in return of GetSMContextByRefInDB() but not used

File: context/db.go [hunk 4]
```diff
@@ -332,10 +338,14 @@ func GetSMContextBySEIDInDB(seidUint uint64) (smContext *SMContext) {
 	filter["seid"] = seid
 
 	result := MongoDBLibrary.RestfulAPIGetOne(SeidSmContextCol, filter)
-	ref := result["ref"].(string)
-	fmt.Println("StoreSeidContextInDB, result string : ", ref)
-
-	return GetSMContext(ref)
+	if result != nil {
+		ref := result["ref"].(string)
+		logger.CtxLog.Debugf("StoreSeidContextInDB, result string : ", ref)
```
Comment (UpadhyayaAnkur) [inline context/db.go:343]: Format specifier missing in log

**Review by UpadhyayaAnkur — APPROVED** [review]

**Review by thakurajayL — DISMISSED** [review]

**Review by UpadhyayaAnkur — APPROVED** [review]

**Pull request merged**